torch/csrc/distributed/c10d/symm_mem/env.hpp

```diff
@@ -0,0 +1,19 @@
+#include <cstdlib>
+#include <cstring>
+
+namespace c10d::symmetric_memory {
+
+static int getenv_nblocks() {
+  static int num_blocks = -1; // Uninitialized
+  if (num_blocks == -1) {
+    const char* str = getenv("TORCH_SYMMMEM_NBLOCKS");
```

Comment:
Don't we have a c10 version that is thread safe?

Reply:
Done!